git-github-introduction.md

```diff
@@ -0,0 +1,359 @@
+
+$ git commit -m "Add missing section to README noticed by reviewer"
+
+$ git push origin master
+```
+
+### Resolve a Merge Conflict
+
+Merge conflicts might occur with Git's push or pull actions if multiple contributors attempt to modify the same section of a file, and merge their conflicting modifications to the server’s “master" branch. For example, suppose one contributor wants "foo" to appear in the first three characters of a file, while another contributor desires "bar". When a merge conflict occurs, the affected contributors must mutually resolve the conflicting section. That is, the file in the previous example must be edited to produce one unique version of the file; either "foo" appears in the first three characters, or "bar" does.
+
+If Git detects a merge conflict after a pull or merge action, it inserts special markers in the conflicting section(s) of the file to indicate what needs to be resolved:
+
+```
+    <<<<<< HEAD
+    foo
+    ======
+    bar
+    >>>>>>
+```
+
+The text between <<<<<< HEAD and ===== is what is currently on the server’s “master” branch.  The text between ===== and >>>>>> is what is being merged and causing the conflict.  To resolve the merge conflict, a contributor must open the file in a text editor and manually remove the merge conflict markers and either keep what is currently on the “master” branch or what is being merged.  In the merge conflict example above, a contributor would keep either the text “foo” or “bar”, and remove the rest of the markers inserted by Git.  To finalize the merge conflict, the contributor must commit the resolved changes.
+
+**Steps to resolve the merge conflict:**
+
+Edit the file containing the merge conflict in a text editor.
+To discard what is currently on the server and replace it with “bar”, delete “foo” and the three inserted markers.
+
+1. Save the file.
+
+2. ```$ git add```
+
+3. ```$ git commit```
+
+4. ```$ git push```
+
+The following page provides a comprehensive guide on resolving merge conflicts: https://help.github.com/articles/resolving-a-merge-conflict-from-the-command-line/
+
+## Miscellaneous Git commands
+Typing **git help -a** on the command-line outputs the full list of commands supported by Git.
+Consult Git's help pages for more information about each command.
+Another resource that covers useful Git commands can be found here:
+http://zackperdue.com/tutorials/super-useful-need-to-know-git-commands
+
+The section below lists some useful, common Git commands.
+
+
+To show changes/differences in the README.txt that have not been committed:
+
+```
+$ git diff README.txt
+diff --git a/README.txt b/README.txt
+index 70a3d0f..93bce69 100644
+--- a/README.txt
++++ b/README.txt
+@@ -1 +1,2 @@
+ README file.
+ +new line added here.
+```
+
+
+To Mark README.txt as removed and stage it for removal:
+
+```
+$ git rm README.txt
+```
+
+
+To unstage the modified README.txt file and reset its contents to the current version on the repository:

```

Comment:
Not sure if this is the behavior of git reset in this case. Doesn't git reset just unstage but leave the contents as they are? You're probably git reset --hard or git checkout -- [file]